EIPS/eip-XXXX.md

```diff
@@ -0,0 +1,91 @@
+---
+eip: XXX
+title: eth/70 - partial block receipt lists
+description: Adds a facility for paginating block receipts in the p2p protocol
+author: Felix Lange <fjl@ethereum.org>, Jochem Brouwer (@jochem-brouwer), Giulio Rebuffo (@Giulio2002)
+discussions-to:
+status: Final
+type: Standards Track
+category: Networking
+created: 2025-06-16
+requires: 7642, 7825
+---
+
+## Abstract
+
+This EIP modifies the 'eth' p2p protocol to allow requesting partial block receipt lists.
+
+## Motivation
+
+As Ethereum moves toward a higher block gas limit on mainnet, the worst-case total size of
+a block receipts list also becomes larger, and may eventually exceed the 10MB message size
+limit commonly applied in clients. This can lead to sync failures.
+
+## Specification
+
+Modify the encoding for receipts in the `Receipts (0x10)` message as follows:
+
+- (eth/69): `[request-id: P, [[receipt₁, receipt₂], ...]]`
+
+- (eth/70): `[request-id: P, lastBlockIncomplete: {0,1}, [[receipt₁, receipt₂], ...]]`
+
+If the `lastBlockIncomplete` flag is set to true (`1`), the last receipt list does not
+contain all receipts of the block, and the client will have to request the remaining
+receipts of that block in a new request.
+
+To support such partial queries, we also modify the `GetReceipts (0x0f)` message:
+
+- (eth/69): `[request-id: P, [blockhash₁: B_32, blockhash₂: B_32, ...]]`
+
+- (eth/70): `[request-id: P, firstBlockReceiptIndex: P, [blockhash₁: B_32, blockhash₂: B_32, ...]]`
+
+For the first block in the list of requested block hashes, the server should omit receipts
```

Comment:
```suggestion
For the first block in the list of requested block hashes, the server must omit receipts
```

Since receipts are unique (due to `cumulativeGasUsed`) this is technically not a problem, but it would make implementations more complex if we have to deal with the case that the server is sending us receipts which we already have (and thus did not request) :thinking: This also helps to find the position (key) where the receipts should enter the trie, otherwise we have to take all receipts which we have gotten, then sort them by `cumulativeGasUsed` and then insert them in the trie to calculate the root.